src/commands/synthetics/reporters/appUpload.ts

```diff
@@ -0,0 +1,70 @@
+import chalk from 'chalk'
+import ora from 'ora'
+
+import {CommandContext} from '../../../helpers/interfaces'
+
+import {AppUploadDetails} from '../interfaces'
+
+import {ICONS} from './constants'
+
+export class AppUploadReporter {
+  private context: CommandContext
+  private spinner?: ora.Ora
+  private startTime: number
+
+  constructor(context: CommandContext) {
+    this.context = context
+    this.startTime = Date.now()
+  }
+
+  public start = (appsToUpload: AppUploadDetails[], prependLineBreak = false): void => {
+    this.write(`${prependLineBreak ? '\n' : ''}${appsToUpload.length} mobile application(s) to upload:\n`)
+    this.write(appsToUpload.map((appToUpload) => this.getAppRepr(appToUpload)).join('\n') + '\n')
+  }
+
+  public renderProgress = (numberOfApplicationsLeft: number): void => {
+    const text = `Uploading ${numberOfApplicationsLeft} application(s)…`
+    this.spinner?.stop()
+    this.spinner = ora({
+      stream: this.context.stdout,
+      text,
+    })
+    this.spinner.start()
+  }
+
+  public reportSuccess = (appendLineBreak = false): void => {
+    this.endRendering()
+    this.write(
+      `${ICONS.SUCCESS} Uploaded applications in ${Math.round((Date.now() - this.startTime) / 1000)}s${
+        appendLineBreak ? '\n' : ''
```

Comment:
Why not always append a line break? 🤔

Reply:
For example, this doesn't look too good IMO (I don't know if `appendLineBreak = true` would have fixed it in this context though):
 
![image](https://github.com/DataDog/datadog-ci/assets/9317502/9bfa0fab-199a-4c76-9ddb-fcd8db410159)